lib/cartopy/crs.py

```diff
@@ -2006,3 +2066,28 @@ def epsg(code):
     """
     import cartopy._epsg
     return cartopy._epsg._EPSGProjection(code)
+
+
+def from_proj4(proj4_terms, globe=None, bounds=None):
+    from cartopy._proj4 import get_proj4_dict, _PROJ4Projection
+    proj4_dict = get_proj4_dict(proj4_terms)
+
+    proj_to_crs = {
```

Comment:
I'd like this to be derived from the projection classes, not a static dictionary, if possible.

My branch does this at https://github.com/pelson/cartopy/compare/b15391d1b794c6fc37f94d0be70e3d1900b0a2ef...pelson:repr_from_proj4_init#diff-2af330a440b389503b4c32a5fbd26471R54 with:

```

+    def all_subclasses(cls):
+        return cls.__subclasses__() + [g for s in cls.__subclasses__()
+                                       for g in all_subclasses(s)]
+
+    potentials = []
+    for cls in all_subclasses(CRS):
+        if getattr(cls, '_proj4_proj', None) == proj:
+            potentials.append(cls)
```

Reply:
Ok, I can do this. I wasn't sure how "hacky" you wanted to get. ;)